soroban-env-host/src/auth.rs

```diff
@@ -312,6 +313,16 @@ struct RecordingAuthInfo {
     disable_non_root_auth: bool,
 }
 
+#[cfg(test)]
+impl std::hash::Hash for RecordingAuthInfo {
```

Comment:
While observing recording auth could have some marginal testing value, I'm not sure it should belong to observations (as almost everything else there concerns only observable protocol changes). So I wouldn't feel bad if we just skipped observing recording auth completely.
FWIW majority of auth tests works in the enforcing mode. We do have test cases that cover recording auth and I think these are good enough to make sure recording auth does hasn't diverged.

Reply:
the test observations are made without recording, it's just to make the hash derives compile

Reply:
I see. Should it be simplified to a constant then, so that we're sure recording auth can never influence observations? Up to your preference though.

Reply:
I think I'd actually prefer to notice if we ever do!